modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache

```diff
@@ -277,8 +277,24 @@ export class {{classname}} extends runtime.BaseAPI {
         {{/isArray}}
         {{/formParams}}
         {{/hasFormParams}}
+
+        let urlPath = `{{{path}}}`;
+        {{#pathParams}}
+        {{#isDateTimeType}}
+        urlPath = urlPath.replace(`{${"{{baseName}}"}}`, encodeURIComponent((requestParameters['{{paramName}}'] as any).toISOString()));
```

Comment:
please also check here whether it actually is a `date` object, i.e. `instanceof Date`

same below

Reply:
If it isn't a date what should we do? Obvious options seem to be to stringify whatever it is, or throw an error in the generated code

Reply:
Also, just to call out, none of the other parameter types are making that check now (query params, form params). Assume that if I add the check I should update all those to have the same handling?

Reply:
yeah if its not a Date, just stringify. not sure this can happen in typescript-fetch, but just to be sure.
the other places (query params) should probably also have some safeguard, but ideally in a separate PR

Reply:
Added type checking to date params: [373a34c](https://github.com/OpenAPITools/openapi-generator/pull/19313/commits/373a34c78ccd35a9550c0ece26dabf9e52a7e843)